src/auth/OAUthWithIDTokenValidation.js

```diff
@@ -0,0 +1,92 @@
+var jwt = require('jsonwebtoken');
+var jwksClient = require('jwks-rsa');
+var Promise = require('bluebird');
+
+var ArgumentError = require('rest-facade').ArgumentError;
+
+/**
+ * @class
+ * Abstracts the `oauth.create` method with additional id_token validation\
+ * @constructor
+ * @memberOf module:auth
+ *
+ * @param  {Object}              options                Authenticator options.
+ * @param  {String}              options.baseUrl        The auth0 account URL.
+ * @param  {String}              options.domain       AuthenticationClient server domain
+ * @param  {String}              [options.clientId]     Default client ID.
+ * @param  {String}              [options.clientSecret] Default client Secret.
+ */
+var OAUthWithIDTokenValidation = function(oauth, options) {
+  if (!oauth) {
+    throw new ArgumentError('Missing authenticator options');
+  }
+
+  if (!options) {
+    throw new ArgumentError('Missing authenticator options');
+  }
+
+  if (typeof options !== 'object') {
+    throw new ArgumentError('The authenticator options must be an object');
+  }
+
+  this.oauth = oauth;
+  this.clientId = options.clientId;
+  this.clientSecret = options.clientSecret;
+  this.domain = options.domain;
+};
+
+/**
+ * Creates an oauth request
+ *
+ * @method    create
+ * @memberOf  module:auth.OAuthWithIDTokenValidation.prototype
+ *
+ * @param   {Object}    options               Options are passed through
+ * @param   {Function}  [callbabck]           Callback function
+ *
+ * @return  {Promise|undefined}
+ */
+OAUthWithIDTokenValidation.prototype.create = function(params, data, cb) {
+  const createAndValidate = this.oauth.create(params, data).then(r => {
+    var _this = this;
+    if (r.id_token) {
+      function getKey(header, callback) {
+        if (header.alg === 'HS256') {
+          return callback(null, Buffer.from(_this.clientSecret, 'base64'));
+        }
+        jwksClient({
+          jwksUri: 'https://' + _this.domain + '/.well-known/jwks.json'
+        }).getSigningKey(header.kid, function(err, key) {
+          if (err) {
+            return callback(err);
```

Comment:
Do you want to return the `callback` or call it? Here you return it, below you call it.